config.go

```diff
@@ -10,137 +10,139 @@ import (
 
 // NodeInfo detailed information from a node
 type NodeInfo struct {
-	system_id     string
-	hostname      string
-	url           string
-	power_state   string
-	cpu_count     uint16
-	architecture  string
-	distro_series string
-	hwe_kernel    string
-	memory        uint64
-	osystem       string
-	status        uint16
-	tag_names     []string
-	data          map[string]interface{}
+	tagNames     []string
+	systemID     string
```

Comment:
They don't, actually - `json.Decode()` can't access the private attributes since it's a different package. Check out https://github.com/Roblox/terraform-provider-maas/blob/master/config.go#L29 - that's the function that does it. All of the new code has the struct tags and uses `json.Decode()` to decode the server's response. The gomaasapi package does something similar to this, but it doesn't export the types, so we can't use em.